compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/ComposeContainerLifecycleOwnerTest.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.window
+
+import androidx.compose.ui.assertThat
+import androidx.compose.ui.isEqualTo
+import androidx.compose.ui.scene.ComposeContainer
+import androidx.lifecycle.Lifecycle
+import app.cash.turbine.test
+import javax.swing.JFrame
+import javax.swing.JLayeredPane
+import javax.swing.SwingUtilities
+import kotlin.test.assertNotNull
+import kotlinx.coroutines.test.runTest
+import org.jetbrains.skiko.SkiaLayerAnalytics
+import org.junit.Test
+
+class ComposeContainerLifecycleOwnerTest {
+    @Test
+    fun allEvents() = runTest {
+        val window = JFrame().apply {
+            isVisible = false
+        }
+        val pane = JLayeredPane().also(window::add)
+        var container: ComposeContainer? = null
+        SwingUtilities.invokeAndWait {
+            container = ComposeContainer(container = pane, skiaLayerAnalytics = SkiaLayerAnalytics.Empty, window = window)
+        }
+        assertNotNull(container)
+
+        container!!.lifecycle.currentStateFlow.test {
```

Comment:
As far as I remember, `currentStateFlow` can skip some intermediate states, so, probably `awaitItem()` from the other thread can sometimes return the wrong state (the latest, instead of intermediate).

Instead, we should test via `addObserver`, it intercepts all states.

Reply:
State conflation is a property of `StateFlow` and depending on the desired behavior, it is possible to write test in such a way that all intermediate states will be correctly delivered. For example, by using `Unconfined` dispatcher.
But that's out of scope for now, as I've rewritten the test to use observer instead.